packages/components/src/components/multi-select/multi-select/multi-select.tsx

```diff
@@ -0,0 +1,334 @@
+  private listElement: HTMLUListElement;
+  private isWithinForm: boolean;
+
+  @Listen('internalOptionUpdate')
+  public updateOptionHandler(): void {
+    if (this.isWithinForm) {
+      updateNativeSelectOptions(this.nativeSelect, this.multiSelectOptions);
+    }
+    this.updateSelectedString();
+  }
+
+  public connectedCallback(): void {
+    this.isWithinForm = isWithinForm(this.host);
+    if (this.isWithinForm) {
+      syncNativeSelect(this.nativeSelect, this.host, this.name, this.disabled, this.required);
+    }
+  }
+
+  public componentWillLoad(): void {
+    document.addEventListener('mousedown', this.onClickOutside, true);
+  }
+
+  public componentDidLoad(): void {
+    this.updateOptions();
+  }
+
+  public componentShouldUpdate(newVal: unknown, oldVal: unknown): boolean {
+    return hasPropValueChanged(newVal, oldVal);
+  }
+
+  public disconnectedCallback(): void {
+    document.removeEventListener('mousedown', this.onClickOutside, true);
+  }
+
+  public render(): JSX.Element {
+    validateProps(this, propTypes);
+    attachComponentCss(
+      this.host,
+      getComponentCss,
+      this.nativeSelect.selectedOptions.length > 0,
+      getDropdownDirection(this.dropdownDirection, this.inputContainer, this.multiSelectOptions),
+      this.isOpen,
+      this.disabled,
+      this.hideLabel,
+      this.state,
+      this.isWithinForm,
+      this.theme
+    );
+    syncMultiSelectOptionProps(this.multiSelectOptions, this.theme);
+
+    const PrefixedTagNames = getPrefixedTagNames(this.host);
+    const dropdownId = 'list';
+
+    return (
+      <Host>
+        <div class="root">
+          <label class="label" id="label">
+            {this.selectedString && (
+              <span class="sr-text">{getSelectedOptions(this.multiSelectOptions).length} options selected</span>
+            )}
+            {!this.hideLabel && hasLabel(this.host, this.label) && (
+              <span class="label__text" onClick={() => this.inputElement.focus()}>
+                {this.label || <slot name="label" />}
+                {isRequiredAndParentNotRequired(this.host, this.nativeSelect) && <Required />}
+              </span>
+            )}
+            {/* TODO: Description should be separated from the label (affects all form components) */}
+            {hasDescription(this.host, this.description) && (
+              <span class="label__text" id="description" onClick={() => this.inputElement.focus()} aria-hidden="true">
+                {this.description || <slot name="description" />}
+              </span>
+            )}
+          </label>
+          <div class={{ 'input-container': true, disabled: this.disabled }} ref={(el) => (this.inputContainer = el)}>
+            <input
+              id="comboxbox"
+              role="combobox"
+              placeholder={this.selectedString || null}
+              autoComplete="off"
+              onInput={this.onFilterChange}
+              onClick={this.onInputClick}
```

Comment:
Perhaps rename to `this.onFilterClick` to be in sync with `this.onFilterChange`.